src/modules/handbook.ts

```diff
@@ -0,0 +1,66 @@
+import { command, default as CookiecordClient, Module } from 'cookiecord';
+import { Message, MessageEmbed } from 'discord.js';
+import algoliasearch from 'algoliasearch/lite';
+import { sendWithMessageOwnership } from '../util/send';
+import { TS_BLUE } from '../env';
+import { decode } from 'html-entities';
+
+const ALGOLIA_APP_ID = 'BH4D9OD16A';
+const ALGOLIA_API_KEY = '3c2db2aef0c7ff26e8911267474a9b2c';
+const ALGOLIA_INDEX_NAME = 'typescriptlang';
+
+const algolia = algoliasearch(ALGOLIA_APP_ID, ALGOLIA_API_KEY, {});
+
+type AlgoliaResult = {
+	hierarchy: Record<string, string | null>;
+	url: string;
+};
+
+export class HandbookModule extends Module {
+	constructor(client: CookiecordClient) {
+		super(client);
+	}
+
+	HANDBOOK_EMBED = new MessageEmbed()
+		.setColor(TS_BLUE)
+		.setTitle('The TypeScript Handbook')
+		.setURL('https://www.typescriptlang.org/docs/handbook/intro.html')
+		.setFooter('You can search with `!handbook <query>`');
+
+	@command({
+		description: 'Search the TypeScript Handbook',
+		aliases: ['hb'],
+		single: true,
+	})
+	async handbook(msg: Message, text: string) {
+		if (!text)
+			return await sendWithMessageOwnership(msg, {
+				embed: this.HANDBOOK_EMBED,
+			});
+		const data = await algolia.search<AlgoliaResult>([
+			{
+				indexName: ALGOLIA_INDEX_NAME,
+				query: text,
+				params: {
+					offset: 0,
+					length: 1,
+				},
+			},
+		]);
+		const hit = data.results[0].hits[0];
+		if (!hit)
+			return await sendWithMessageOwnership(
+				msg,
+				':x: No results found for that query',
+			);
+		const hierarchyParts = [0, 1, 2, 3, 4, 5, 6]
+			.map(i => hit.hierarchy[`lvl${i}`])
+			.filter(x => x);
```

Comment:
Is this needed?

Reply:
What part are you referring to? The structure of the object is:
```json
{
  "lvl0": "Documentation",
  "lvl0": "Documentation",
  "lvl0": "Documentation",
  "lvl0": "Documentation",

Reply:
Sorry, was referring to the .filter(x => x)

Reply:
Ahh my bad, sorry, way too tired to be doing code reviews 😅

Reply:
I can remove the `.map` and make it `[hit.hierarchy.hit0, hit.hierarchy.hit1, ...]`, but this seemed DRYer to me.